command/s3_put.go

```diff
@@ -265,8 +266,12 @@ retryLoop:
 					// if we encounter a problem uploading a file, then we should:
 					//   - skip if it's a single update, not marked optional, and the local file doesn't exist
 					// retry in all other cases
-					if !s3pc.isMulti() && !s3pc.Optional && os.IsNotExist(err) {
-						return errors.Wrapf(err, "missing file %s", fpath)
+					if os.IsNotExist(err) {
+						if s3pc.isMulti() || s3pc.Optional {
+							continue uploadLoop
```

Comment:
We'll still continue the retryLoop on the last file (if it's multi/optional).

Reply:
I think I fixed it in the next patch

Reply:
I think that the last file of a multiupload will still trigger the retry loop.